packages/core/src/State.ts

```diff
@@ -293,11 +299,13 @@ export class State<
   public matches<TSV extends TTypestate['value']>(
     parentStateValue: TSV
   ): this is State<
-    (TTypestate extends any
-      ? { value: TSV; context: any } extends TTypestate
-        ? TTypestate
-        : never
-      : never)['context'],
+    Equals<TTypestate, any> extends 1
```

Comment:
This is not strictly necessary because the other. simple, fix already fixes the reported issue.

This seems a little bit heavy (adding more local complexity here) but handles "more" scenarios. I feel like if the `TTypestate` is not configured then the most sensible approach is to prefer using `TContext`. However, this still doesn't handle `unknown` and `never` and if we do this then it would probably be the most sensible approach for all of those special types. 

I think I'm leaning towards removing this part of the fix for the time being. Thoughts?

Reply:
Yeah, let's remove this and revisit later 👍